### PR TITLE
Honor task specific config

### PR DESCRIPTION
**Pull request by notpron**

### CODE REVIEW

File: src/Shell/Task/QueueTask.php [hunk 1]
```diff
@@ -30,20 +30,6 @@ abstract class QueueTask extends Shell implements QueueTaskInterface
...
-     */
-    public $timeout = 120;
-
-    /**
-     * Number of times a failed instance of this task should be restarted before giving up.
-     *
-     * @var int
-     */
-    public $retries = 1;
-
```
Comment (tersmitten) [inline src/Shell/Task/QueueTask.php]: Where do the defaults come from (now)?

Comment (notpron) [reply src/Shell/Task/QueueTask.php]: From Config.php defaultWorkerTimeout() and defaultWorkerRetries(). Value is found by:

1. check task for a value (if property exists, use that)
2. check config file (Queue.defaultWorkerTimeout and Queue.defaultWorkerRetries)
3. default to had coded value in Config.php (2 minutes, 4 retries)

File: src/Model/Table/QueuedTasksTable.php [hunk 2]
```diff
@@ -452,17 +452,45 @@ public function rerun($taskName): int
...
-            return;
+        $conditions = [];
+
+        // Generate the job specific conditions
+        foreach ($capabilities as $task) {
+            list ($plugin, $name) = pluginSplit($task['name']);
+            $conditions['OR'][] = [
+                'task' => $name,
+                'completed <' => date('Y-m-d H:i:s', time() - (int)$task['cleanupTimeout'])
+            ];
```
Comment (tersmitten) [inline src/Model/Table/QueuedTasksTable.php]: Shall we put the `deleteAll` in the foreach right away?

File: src/Model/Table/QueuedTasksTable.php [hunk 3]
```diff
@@ -452,17 +452,45 @@ public function rerun($taskName): int
...
+    {
+        $conditions = [];
+
+        // Generate the job specific conditions.
+        foreach ($capabilities as $task) {
+            list ($plugin, $name) = pluginSplit($task['name']);
+            $conditions['OR'][] = [
+                'task' => $name,
+                'failed_count >' => $task['retries']
+            ];
```
Comment (tersmitten) [inline src/Model/Table/QueuedTasksTable.php]: Shall we put the `deleteAll` in the foreach right away?

**Pull request merged**